graph-subscriptions-api/src/schema.rs

```diff
@@ -188,10 +278,22 @@ impl RequestTicketDto {
             return Err(AuthError::Unauthorized.into());
         }
         let ticket_payload = ticket_payload_wrapper.unwrap();
-        let _sub = &ticket_payload.active_subscription;
+        let sub = &ticket_payload.active_subscription;
+
+        let schema_ctx = ctx
+            .data_unchecked::<Arc<Mutex<GraphSubscriptionsSchemaCtx>>>()
+            .lock()
+            .await;
 
-        // TODO: BUILD OUT VolumeEstimator LOGIC FROM gateway TO CALCULATE HOW MANY QUERIES AVAILABLE ON THE SUB
-        Ok(0.00)
+        let sub_tier = schema_ctx.subscription_tiers.tier_for_rate(sub.rate);
+        let sub_queries_available =
+            sub_tier.query_rate_limit as i64 * ((sub.end - sub.start).num_seconds());
+        let percentage_used = if sub_queries_available == 0 || sub_queries_available == 0 {
```

Comment:
```suggestion
        let percentage_used = sub_queries_available == 0 {
```

Reply:
Also, can we reuse `est_queries_available` here?
```rust
    async fn query_rate_used_percentage<'ctx>(&self, ctx: &Context<'ctx>) -> Result<f64> {
        let sub_queries_available = self.est_queries_available(ctx).await?;
        let percentage_used = if sub_queries_available == 0 {
            0.00_f64
        } else {
            self.total_query_count as f64 / sub_queries_available as f64
        };
        Ok(percentage_used)
    }
```

Reply:
Yes we can. I wasn't sure how it would work since it is a derived field. But is still a fn that can be resolved. Good catch